Watching for file changes with StatReloader
Processing request for courses: [{'course_type': 'ENGG', 'course_code': '2400'}]
Found 36 sections for ENGG2400 in 0.018s
Generated 36 combinations in 0.000s
Found 36 conflict-free schedules in 0.103s
Processing request for courses: [{'course_type': 'ENGG', 'course_code': '2400'}, {'course_type': 'ENGG', 'course_code': '1410'}]
Found 36 sections for ENGG2400 in 0.018s
Found 70 sections for ENGG1410 in 0.006s
Generated 2520 combinations in 0.001s
Checked 1000/2520 combinations...
Checked 2000/2520 combinations...
Found 1690 conflict-free schedules in 0.474s
Processing request for courses: [{'course_type': 'ENGG', 'course_code': '2400'}, {'course_type': 'ENGG', 'course_code': '1410'}, {'course_type': 'CIS', 'course_code': '2750'}]
Found 36 sections for ENGG2400 in 0.008s
Found 70 sections for ENGG1410 in 0.005s
Found 17 sections for CIS2750 in 0.004s
Generated 42840 combinations in 0.014s
Checked 1000/42840 combinations...
Checked 2000/42840 combinations...
Checked 3000/42840 combinations...
Checked 4000/42840 combinations...
Checked 5000/42840 combinations...
Checked 6000/42840 combinations...
Checked 7000/42840 combinations...
Checked 8000/42840 combinations...
Checked 9000/42840 combinations...
Checked 10000/42840 combinations...
Checked 11000/42840 combinations...
Checked 12000/42840 combinations...
Checked 13000/42840 combinations...
Checked 14000/42840 combinations...
Checked 15000/42840 combinations...
Checked 16000/42840 combinations...
Checked 17000/42840 combinations...
Checked 18000/42840 combinations...
Checked 19000/42840 combinations...
Checked 20000/42840 combinations...
Checked 21000/42840 combinations...
Checked 22000/42840 combinations...
Checked 23000/42840 combinations...
Checked 24000/42840 combinations...
Checked 25000/42840 combinations...
Checked 26000/42840 combinations...
Checked 27000/42840 combinations...
Checked 28000/42840 combinations...
Checked 29000/42840 combinations...
Checked 30000/42840 combinations...
Checked 31000/42840 combinations...
Checked 32000/42840 combinations...
Checked 33000/42840 combinations...
Checked 34000/42840 combinations...
Checked 35000/42840 combinations...
Checked 36000/42840 combinations...
Checked 37000/42840 combinations...
Checked 38000/42840 combinations...
Checked 39000/42840 combinations...
Checked 40000/42840 combinations...
Checked 41000/42840 combinations...
Checked 42000/42840 combinations...
Found 1170 conflict-free schedules in 11.743s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Processing request for courses: [{'course_type': 'CIS', 'course_code': '2750'}]
Course CIS2750: 17 sections with events
Found 17 conflict-free schedules in 0.065s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Processing request for courses: [{'course_type': 'CIS', 'course_code': '2750'}]
Course CIS2750: 17 sections with events
Found 17 conflict-free schedules in 0.068s
Processing request for courses: [{'course_type': 'CIS', 'course_code': '2750'}, {'course_type': 'ENGG', 'course_code': '2400'}]
Course CIS2750: 17 sections with events
Course ENGG2400: 36 sections with events
Found 457 conflict-free schedules in 0.140s
Processing request for courses: [{'course_type': 'CIS', 'course_code': '2750'}, {'course_type': 'ENGG', 'course_code': '2400'}, {'course_type': 'ENGG', 'course_code': '1410'}]
Course CIS2750: 17 sections with events
Course ENGG2400: 36 sections with events
Course ENGG1410: 70 sections with events
Found 11151 conflict-free schedules in 0.473s
Processing request for courses: [{'course_type': 'CIS', 'course_code': '2750'}, {'course_type': 'ENGG', 'course_code': '2400'}, {'course_type': 'ENGG', 'course_code': '1410'}]
Course CIS2750: 17 sections with events
Course ENGG2400: 36 sections with events
Course ENGG1410: 70 sections with events
Found 11151 conflict-free schedules in 0.372s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CIS2750: 17 sections with events
Course ENGG1410: 70 sections with events
Course ENGG2400: 36 sections with events
Returned 100 schedules out of 100 found in 0.31s
Course CIS2750: 17 sections with events
Course ENGG1410: 70 sections with events
Course ENGG2400: 36 sections with events
Returned 100 schedules out of 100 found in 0.26s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CIS2750: 17 sections with events
Course ENGG1410: 70 sections with events
Course ENGG2400: 36 sections with events
Found 100 conflict-free schedules in 0.27s
Returned 100 schedules out of 100 found in 0.28s
Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\coursescheduler\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\coursescheduler\settings.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\apps.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /gpacalc
Not Found: /gpacalc/
Not Found: /gpacalc/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\coursescheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /gpacalc/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\urls.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_course_codes' not found. 'get_course_codes' is not a valid view function or pattern name.
Not Found: /favicon.ico
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\models.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Not Found: /submit_suggestion/
Not Found: /submit_suggestion/
Watching for file changes with StatReloader
Not Found: /submit_suggestion/
Not Found: /api/submit_suggestion/
Watching for file changes with StatReloader
Not Found: /get_course_codes/
Not Found: /get_course_codes/
Not Found: /search/
Not Found: /search/
Not Found: /search/
Not Found: /search/
Not Found: /search/
Watching for file changes with StatReloader
Not Found: /search/
Not Found: /search/
Not Found: /search/
Method Not Allowed (GET): /api/scheduler/search/
Method Not Allowed (GET): /api/scheduler/search/
Not Found: /search/
Not Found: /add_to_calendar/
Unauthorized: /api/auth/user/
Watching for file changes with StatReloader
Not Found: /conflictfree/
Not Found: /conflictfree/
Not Found: /conflictfree/
Not Found: /conflictfree/
Not Found: /conflictfree/
Not Found: /conflictfree/
Unauthorized: /api/auth/user/
Not Found: /conflictfree/
Not Found: /conflictfree
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /scheduler/conflictfree/
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /conflictfree/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\coursescheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /conflictfree/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\coursescheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /conflict_test/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\coursescheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /conflict_test/
Not Found: /conflict_test/
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /scheduler/conflictfree/
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /scheduler/conflictfree/
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /scheduler/conflictfree/
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /scheduler/conflictfree/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: Expecting value: line 1 column 1 (char 0)
Internal Server Error: /scheduler/conflictfree/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\urls.py changed, reloading.
Watching for file changes with StatReloader
Not Found: /api/course_types/
Course CIS2750: 16 sections with events
Found 16 conflict-free schedules in 0.04s
Returned 16 schedules out of 16 found in 0.04s
Course CIS2750: 16 sections with events
Course ENGG1410: 70 sections with events
Found 100 conflict-free schedules in 0.10s
Returned 100 schedules out of 100 found in 0.10s
Course CIS2750: 16 sections with events
Course ENGG1410: 70 sections with events
Course ENGG2400: 36 sections with events
Found 100 conflict-free schedules in 0.14s
Returned 100 schedules out of 100 found in 0.14s
Course CIS2750: 16 sections with events
Course ENGG1410: 70 sections with events
Course ENGG2400: 36 sections with events
Course ENGG1410: 70 sections with events
Found 100 conflict-free schedules in 0.21s
Returned 100 schedules out of 100 found in 0.21s
Course CIS2750: 16 sections with events
Course CIS2750: 16 sections with events
Found 100 conflict-free schedules in 0.04s
Returned 100 schedules out of 100 found in 0.04s
Course CIS2750: 16 sections with events
Found 16 conflict-free schedules in 0.02s
Returned 16 schedules out of 16 found in 0.02s
Watching for file changes with StatReloader
Course CISCIS*2750: 17 sections with events
Found 17 conflict-free schedules in 0.03s
Returned 17 schedules out of 17 found in 0.03s
Course CISCIS*2750: 17 sections with events
Course ENGGENGG*1410: 70 sections with events
Course ENGGENGG*2400: 36 sections with events
Course ENGGENGG*2410: 48 sections with events
Found 100 conflict-free schedules in 0.36s
Returned 100 schedules out of 100 found in 0.36s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\api_urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\api_urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\api_urls.py changed, reloading.
Watching for file changes with StatReloader
Forbidden (CSRF cookie not set.): /api/scheduler/course_types/
Forbidden (CSRF cookie not set.): /api/scheduler/course_types/
Forbidden (CSRF cookie not set.): /api/scheduler/course_types/
Watching for file changes with StatReloader
Forbidden (CSRF cookie not set.): /api/scheduler/course_types/
Forbidden (CSRF cookie not set.): /api/scheduler/course_types/
Method Not Allowed (POST): /api/scheduler/course_types/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Method Not Allowed (POST): /api/scheduler/course_types/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CISCIS*2750: 17 sections with events
Found 17 conflict-free schedules in 0.04s
Returned 17 schedules out of 17 found in 0.04s
Course ENGGENGG*1410: 70 sections with events
Found 70 conflict-free schedules in 0.11s
Returned 70 schedules out of 70 found in 0.11s
Course CISCIS*2750: 17 sections with events
Found 17 conflict-free schedules in 0.02s
Returned 17 schedules out of 17 found in 0.02s
Watching for file changes with StatReloader
Course CISCIS*2750: 1 sections with events
Found 1 conflict-free schedules in 0.02s
Returned 1 schedules out of 1 found in 0.02s
Course ENGGENGG*1410: 70 sections with events
Found 70 conflict-free schedules in 0.08s
Returned 70 schedules out of 70 found in 0.08s
Course CISCIS*2750: 16 sections with events
Found 16 conflict-free schedules in 0.02s
Returned 16 schedules out of 16 found in 0.02s
Course CISCIS*2750: 1 sections with events
Found 1 conflict-free schedules in 0.01s
Returned 1 schedules out of 1 found in 0.01s
Course CISCIS*2750: 1 sections with events
Course ENGGENGG*1410: 70 sections with events
Course ENGGENGG*2400: 36 sections with events
Course ENGGENGG*2410: 48 sections with events
Found 100 conflict-free schedules in 0.17s
Returned 100 schedules out of 100 found in 0.17s
Course ENGGENGG*6090: 3 sections with events
Found 3 conflict-free schedules in 0.01s
Returned 3 schedules out of 3 found in 0.01s
Course ENGGENGG*4460: 18 sections with events
Found 18 conflict-free schedules in 0.04s
Returned 18 schedules out of 18 found in 0.04s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\models.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Course ENGGENGG*1410: 70 sections with events
Course CHEMCHEM*1040: 126 sections with events
Found 100 conflict-free schedules in 0.21s
Returned 100 schedules out of 100 found in 0.22s
Course CHEMCHEM*1040: 126 sections with events
Found 100 conflict-free schedules in 0.19s
Returned 100 schedules out of 100 found in 0.19s
Course ENGGENGG*1410: 70 sections with events
Found 70 conflict-free schedules in 0.13s
Returned 70 schedules out of 70 found in 0.13s
Course CHEMCHEM*1040: 24 sections with events
Found 24 conflict-free schedules in 0.04s
Returned 24 schedules out of 24 found in 0.04s
Course CHEMCHEM*1040: 126 sections with events
Found 100 conflict-free schedules in 0.14s
Returned 100 schedules out of 100 found in 0.14s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CHEMCHEM*1040: 126 sections with events
Found 100 conflict-free schedules in 0.15s
Returned 100 schedules out of 100 found in 0.15s
Course CHEMCHEM*1040: 126 sections with events
Found 100 conflict-free schedules in 0.13s
Returned 100 schedules out of 100 found in 0.13s
Course CHEMCHEM*1040: 126 sections with events
Course CISCIS*2750: 1 sections with events
Found 100 conflict-free schedules in 0.14s
Returned 100 schedules out of 100 found in 0.14s
Course CISCIS*2750: 1 sections with events
Found 1 conflict-free schedules in 0.02s
Returned 1 schedules out of 1 found in 0.02s
Course CISCIS*2750: 1 sections with events
Course ENGGENGG*2400: 36 sections with events
Found 36 conflict-free schedules in 0.05s
Returned 36 schedules out of 36 found in 0.05s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CHEMCHEM*1040: 126 sections with events
Found 100 conflict-free schedules in 0.13s
Returned 100 schedules out of 100 found in 0.13s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CHEMCHEM*1040: 126 sections with events
Found 126 conflict-free schedules in 0.18s
Returned 100 schedules out of 126 found in 0.18s
Course CHEMCHEM*1040: 126 sections with events
Course CISCIS*2750: 1 sections with events
Course ENGGENGG*1410: 70 sections with events
Course ENGGENGG*2400: 36 sections with events
Course ENGGENGG*2410: 48 sections with events
Course ENGGENGG*4460: 18 sections with events
Watching for file changes with StatReloader
Course CHEMCHEM*1040: 126 sections with events
Course CISCIS*2750: 1 sections with events
Course ENGGENGG*1410: 70 sections with events
Course ENGGENGG*2400: 36 sections with events
Course ENGGENGG*2410: 48 sections with events
Course ENGGENGG*4460: 18 sections with events
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CHEMCHEM*1040: 126 sections with events
Course CISCIS*2750: 1 sections with events
Course ENGGENGG*1410: 70 sections with events
Course ENGGENGG*2400: 36 sections with events
Course ENGGENGG*2410: 48 sections with events
Course ENGGENGG*4460: 18 sections with events
Watching for file changes with StatReloader
Course CHEM1040: 126 sections with events
Found 126 conflict-free schedules in 0.11s
Returned 100 schedules out of 126 found in 0.11s
Course CHEM1040: 126 sections with events
Course ECON1050: 5 sections with events
Found 630 conflict-free schedules in 0.17s
Returned 100 schedules out of 630 found in 0.17s
Course CHEM1040: 126 sections with events
Course ECON1050: 5 sections with events
Course ENGG1410: 70 sections with events
Course ENGG2400: 36 sections with events
Course ENGG2410: 48 sections with events
Watching for file changes with StatReloader
Course CHEM1040: 126 sections with events
Found 126 conflict-free schedules in 0.17s
Returned 100 schedules out of 126 found in 0.17s
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.02s
Returned 5 schedules out of 5 found in 0.02s
Course ECON1050: 5 sections with events
Course ENGG1410: 70 sections with events
Found 350 conflict-free schedules in 0.10s
Returned 100 schedules out of 350 found in 0.10s
Course ECON1050: 5 sections with events
Course ENGG2400: 36 sections with events
Found 180 conflict-free schedules in 0.06s
Returned 100 schedules out of 180 found in 0.06s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ECON1050: 5 sections with events
Course ENGG2400: 36 sections with events
Found 180 conflict-free schedules in 0.07s
Returned 100 schedules out of 180 found in 0.07s
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.01s
Returned 5 schedules out of 5 found in 0.01s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.02s
Returned 5 schedules out of 5 found in 0.02s
Course ECON1050: 5 sections with events
Course CHEM1040: 126 sections with events
Found 630 conflict-free schedules in 0.18s
Returned 100 schedules out of 630 found in 0.18s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ECON1050: 5 sections with events
Course CHEM1040: 126 sections with events
Found 630 conflict-free schedules in 0.21s
Returned 100 schedules out of 630 found in 0.21s
Watching for file changes with StatReloader
Course CHEM1040: 126 sections with events
Found 126 conflict-free schedules in 0.25s
Returned 100 schedules out of 126 found in 0.25s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ENGG1410: 70 sections with events
Found 70 conflict-free schedules in 0.14s
Returned 70 schedules out of 70 found in 0.14s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ENGG1410: 70 sections with events
Found 70 conflict-free schedules in 0.25s
Returned 70 schedules out of 70 found in 0.25s
Watching for file changes with StatReloader
Course CIS2750: 16 sections with events
Found 16 conflict-free schedules in 0.08s
Returned 16 schedules out of 16 found in 0.08s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CIS2750: 16 sections with events
Found 16 conflict-free schedules in 0.05s
Returned 16 schedules out of 16 found in 0.05s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CIS2750: 1 sections with events
Found 1 conflict-free schedules in 0.02s
Returned 1 schedules out of 1 found in 0.02s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CIS2750: 1 sections with events
Found 1 conflict-free schedules in 0.01s
Returned 1 schedules out of 1 found in 0.01s
Watching for file changes with StatReloader
Course CIS2750: 16 sections with events
Found 16 conflict-free schedules in 0.05s
Returned 16 schedules out of 16 found in 0.05s
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.02s
Returned 5 schedules out of 5 found in 0.02s
Watching for file changes with StatReloader
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.02s
Returned 5 schedules out of 5 found in 0.02s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.03s
Returned 5 schedules out of 5 found in 0.03s
Course ECON1050: 5 sections with events
Found 5 conflict-free schedules in 0.01s
Returned 5 schedules out of 5 found in 0.01s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: 'NoneType' object has no attribute 'split'
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: 'NoneType' object has no attribute 'split'
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: 'NoneType' object has no attribute 'split'
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: 'NoneType' object has no attribute 'split'
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: 'NoneType' object has no attribute 'split'
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: not enough values to unpack (expected 2, got 1)
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: name 'prin' is not defined
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: not enough values to unpack (expected 2, got 1)
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course CHEM1040: 126 sections with events
Found 126 conflict-free schedules in 0.19s
Returned 100 schedules out of 126 found in 0.19s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Error in conflict_free_schedule: not enough values to unpack (expected 2, got 1)
Internal Server Error: /api/scheduler/conflict_free_schedule/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ECON1050: 1 sections with events
Found 1 conflict-free schedules in 0.02s
Returned 1 schedules out of 1 found in 0.02s
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\views.py changed, reloading.
Watching for file changes with StatReloader
Course ECON1050: 1 sections with events
Found 1 conflict-free schedules in 0.02s
Returned 1 schedules out of 1 found in 0.02s
Course ECON1050: 1 sections with events
Found 1 conflict-free schedules in 0.01s
Returned 1 schedules out of 1 found in 0.01s
Course ECON1050: 1 sections with events
Found 1 conflict-free schedules in 0.03s
Returned 1 schedules out of 1 found in 0.03s
Course ECON1050: 1 sections with events
Found 1 conflict-free schedules in 0.02s
Returned 1 schedules out of 1 found in 0.02s
Course CHEM1040: 126 sections with events
Course CIS2750: 1 sections with events
Found 95 conflict-free schedules in 0.35s
Returned 95 schedules out of 95 found in 0.35s
Course CHEM1040: 126 sections with events
Course CIS2750: 1 sections with events
Found 95 conflict-free schedules in 0.33s
Returned 95 schedules out of 95 found in 0.33s
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 144, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 145, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 145, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 145, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\models.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 145, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 145, in calculate_gpa
    course_obj = Course.objects.get(
                 ^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 636, in get
    raise self.model.MultipleObjectsReturned(
scheduler.models.Course.MultipleObjectsReturned: get() returned more than one Course -- it returned 2!
Watching for file changes with StatReloader
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\api_urls.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\api_urls.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'api-gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'api-gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'api-gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'api-gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'api-gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'api-gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'api-gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'api-gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'gpacalc-api_url'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'gpacalc-api_url' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'api-gpacalc'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'api-gpacalc' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'api-gpacalc'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'api-gpacalc' is not a registered namespace
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\api_urls.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_offered_terms' not found. 'get_offered_terms' is not a valid view function or pattern name.
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\api_urls.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 81, in reverse
    extra, resolver = resolver.namespace_dict[ns]
                      ~~~~~~~~~~~~~~~~~~~~~~~^^^^
KeyError: 'gpacalc-api-'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 92, in reverse
    raise NoReverseMatch("%s is not a registered namespace" % key)
django.urls.exceptions.NoReverseMatch: 'gpacalc-api-' is not a registered namespace
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 160, in calculate_gpa
    offered_term = c["offered_term"],
                   ~^^^^^^^^^^^^^^^^
KeyError: 'offered_term'
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
Internal Server Error: /gpacalc/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 26, in index
    return render(request, "gpacalc/index.html", {
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\shortcuts.py", line 25, in render
    content = loader.render_to_string(template_name, context, request, using=using)
              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\loader.py", line 62, in render_to_string
    return template.render(context, request)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\backends\django.py", line 107, in render
    return self.template.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 171, in render
    return self._render(context)
           ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 163, in _render
    return self.nodelist.render(context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in render
    return SafeString("".join([node.render_annotated(context) for node in self]))
                              ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 1016, in <listcomp>
    return SafeString("".join([node.render_annotated(context) for node in self]))
                               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\base.py", line 977, in render_annotated
    return self.render(context)
           ^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\template\defaulttags.py", line 480, in render
    url = reverse(view_name, args=args, kwargs=kwargs, current_app=current_app)
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\base.py", line 98, in reverse
    resolved_url = resolver._reverse_with_prefix(view, prefix, *args, **kwargs)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\urls\resolvers.py", line 831, in _reverse_with_prefix
    raise NoReverseMatch(msg)
django.urls.exceptions.NoReverseMatch: Reverse for 'get_course_types' not found. 'get_course_types' is not a valid view function or pattern name.
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\api_urls.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/course_events/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 115, in get_course_events
    evs = CourseEvent.objects.filter(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1493, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1511, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1518, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1646, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1678, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1526, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1333, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1805, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'course_offered_term' into field. Choices are: assessmentgrade, course, course_id, days, description, end_date, event_date, event_type, id, location, start_date, time, weightage
Internal Server Error: /api/gpacalc/course_events/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 115, in get_course_events
    evs = CourseEvent.objects.filter(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1493, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1511, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1518, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1646, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1678, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1526, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1333, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1805, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'course_offered_term' into field. Choices are: assessmentgrade, course, course_id, days, description, end_date, event_date, event_type, id, location, start_date, time, weightage
Internal Server Error: /api/gpacalc/course_events/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 115, in get_course_events
    evs = CourseEvent.objects.filter(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1493, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1511, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1518, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1646, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1678, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1526, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1333, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1805, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'course_offered_term' into field. Choices are: assessmentgrade, course, course_id, days, description, end_date, event_date, event_type, id, location, start_date, time, weightage
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 174, in calculate_gpa
    offered_term = c["offered_term"],
                   ~^^^^^^^^^^^^^^^^
KeyError: 'offered_term'
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/course_events/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 115, in get_course_events
    evs = CourseEvent.objects.filter(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1493, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1511, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1518, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1646, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1678, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1526, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1333, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1805, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'course_offered_term' into field. Choices are: assessmentgrade, course, course_id, days, description, end_date, event_date, event_type, id, location, start_date, time, weightage
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/course_events/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\csrf.py", line 65, in _view_wrapper
    return view_func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 116, in get_course_events
    evs = CourseEvent.objects.filter(
          ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1493, in filter
    return self._filter_or_exclude(False, args, kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1511, in _filter_or_exclude
    clone._filter_or_exclude_inplace(negate, args, kwargs)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1518, in _filter_or_exclude_inplace
    self._query.add_q(Q(*args, **kwargs))
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1646, in add_q
    clause, _ = self._add_q(q_object, can_reuse)
                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1678, in _add_q
    child_clause, needed_inner = self.build_filter(
                                 ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1526, in build_filter
    lookups, parts, reffed_expression = self.solve_lookup_type(arg, summarize)
                                        ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1333, in solve_lookup_type
    _, field, _, lookup_parts = self.names_to_path(lookup_splitted, self.get_meta())
                                ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\query.py", line 1805, in names_to_path
    raise FieldError(
django.core.exceptions.FieldError: Cannot resolve keyword 'course_offered_term' into field. Choices are: assessmentgrade, course, course_id, days, description, end_date, event_date, event_type, id, location, start_date, time, weightage
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 174, in calculate_gpa
    offered_term = c["offered_term"],
                   ~^^^^^^^^^^^^^^^^
KeyError: 'offered_term'
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 174, in calculate_gpa
    offered_term = c["offered_term"],
                   ~^^^^^^^^^^^^^^^^
KeyError: 'offered_term'
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 191, in calculate_gpa
    AssessmentGrade.objects.create(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 663, in create
    obj.save(force_insert=True, using=self.db)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 902, in save
    self.save_base(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1008, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1169, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1210, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1868, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\compiler.py", line 1882, in execute_sql
    cursor.execute(sql, params)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
django.db.utils.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 191, in calculate_gpa
    AssessmentGrade.objects.create(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 663, in create
    obj.save(force_insert=True, using=self.db)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 902, in save
    self.save_base(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1008, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1169, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1210, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1868, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\compiler.py", line 1882, in execute_sql
    cursor.execute(sql, params)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
django.db.utils.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 191, in calculate_gpa
    AssessmentGrade.objects.create(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 663, in create
    obj.save(force_insert=True, using=self.db)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 902, in save
    self.save_base(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1008, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1169, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1210, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1868, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\compiler.py", line 1882, in execute_sql
    cursor.execute(sql, params)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
django.db.utils.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 191, in calculate_gpa
    AssessmentGrade.objects.create(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 663, in create
    obj.save(force_insert=True, using=self.db)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 902, in save
    self.save_base(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1008, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1169, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1210, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1868, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\compiler.py", line 1882, in execute_sql
    cursor.execute(sql, params)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
django.db.utils.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
pymysql.err.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 191, in calculate_gpa
    AssessmentGrade.objects.create(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 663, in create
    obj.save(force_insert=True, using=self.db)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 902, in save
    self.save_base(
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1008, in save_base
    updated = self._save_table(
              ^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1169, in _save_table
    results = self._do_insert(
              ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\base.py", line 1210, in _do_insert
    return manager._insert(
           ^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\manager.py", line 87, in manager_method
    return getattr(self.get_queryset(), name)(*args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\query.py", line 1868, in _insert
    return query.get_compiler(using=using).execute_sql(returning_fields)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\models\sql\compiler.py", line 1882, in execute_sql
    cursor.execute(sql, params)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 122, in execute
    return super().execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 79, in execute
    return self._execute_with_wrappers(
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 92, in _execute_with_wrappers
    return executor(sql, params, many, context)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 100, in _execute
    with self.db.wrap_database_errors:
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\utils.py", line 91, in __exit__
    raise dj_exc_value.with_traceback(traceback) from exc_value
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\utils.py", line 105, in _execute
    return self.cursor.execute(sql, params)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\db\backends\mysql\base.py", line 76, in execute
    return self.cursor.execute(query, args)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 153, in execute
    result = self._query(query)
             ^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\cursors.py", line 322, in _query
    conn.query(q)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 563, in query
    self._affected_rows = self._read_query_result(unbuffered=unbuffered)
                          ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 825, in _read_query_result
    result.read()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 1199, in read
    first_packet = self.connection._read_packet()
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\connections.py", line 775, in _read_packet
    packet.raise_for_error()
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\protocol.py", line 219, in raise_for_error
    err.raise_mysql_exception(self._data)
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\pymysql\err.py", line 150, in raise_mysql_exception
    raise errorclass(errno, errval)
django.db.utils.DataError: (1264, "Out of range value for column 'achieved_percentage' at row 1")
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\scheduler\models.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 203, in calculate_gpa
    "credits": float(cg.course__credits)
                     ^^^^^^^^^^^^^^^^^^
AttributeError: 'CourseGrade' object has no attribute 'course__credits'
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 203, in calculate_gpa
    "credits": float(cg.coursegrade__credits)
                     ^^^^^^^^^^^^^^^^^^^^^^^
AttributeError: 'CourseGrade' object has no attribute 'coursegrade__credits'
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 209, in calculate_gpa
    total_weighted_percentage += float(cg.final_percentage) * float(cg.credits)
                                                                    ^^^^^^^^^^
AttributeError: 'CourseGrade' object has no attribute 'credits'
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 209, in calculate_gpa
    total_weighted_percentage += float(cg.final_percentage) * float(cg.credits)
                                                                    ^^^^^^^^^^
AttributeError: 'CourseGrade' object has no attribute 'credits'
Internal Server Error: /api/gpacalc/calculate/
Traceback (most recent call last):
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\exception.py", line 55, in inner
    response = get_response(request)
               ^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\core\handlers\base.py", line 197, in _get_response
    response = wrapped_callback(request, *callback_args, **callback_kwargs)
               ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\prg\miniconda\envs\coursescheduler\Lib\site-packages\django\views\decorators\http.py", line 64, in inner
    return func(request, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py", line 209, in calculate_gpa
    total_weighted_percentage += float(cg.final_percentage) * float(cg.credits)
                                                                    ^^^^^^^^^^
AttributeError: 'CourseGrade' object has no attribute 'credits'
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\models.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Unauthorized: /api/auth/user/
Unauthorized: /api/auth/user/
Unauthorized: /api/auth/user/
Unauthorized: /api/auth/user/
Unauthorized: /api/auth/user/
Unauthorized: /api/auth/user/
Watching for file changes with StatReloader
Not Found: /gpacalc/api/gpacalc/progress_export_excel/
Not Found: /gpacalc/api/gpacalc/progress_export_excel/
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
C:\Users\brijesh.thakrar\Downloads\coursescheduler\backend\gpacalc\views.py changed, reloading.
Watching for file changes with StatReloader
